core/txpool/txpool.go

```diff
@@ -2056,3 +2110,24 @@ func (t *lookup) RemotesBelowTip(threshold *big.Int) types.Transactions {
 func numSlots(tx *types.Transaction) int {
 	return int((tx.Size() + txSlotSize - 1) / txSlotSize)
 }
+
+var _ gossip.Tx = (*MempoolTx)(nil)
+
+type MempoolTx struct {
+	Tx *types.Transaction
+}
+
+func (tx *MempoolTx) Hash() []byte {
+	return tx.Hash()
+}
+
+func (tx *MempoolTx) Marshal() ([]byte, error) {
+	return tx.Tx.MarshalBinary()
+}
+
+func (tx *MempoolTx) Unmarshal(b []byte) error {
+	tx.Tx = &types.Transaction{}
+	err := tx.Tx.UnmarshalBinary(b)
+
+	return err
+}
```

Comment:
is there a way to avoid adding this code into `core/txpool/` so that it can be better separated from the geth code?

Reply:
ex. I wonder if specifying an `onAdd` hook inside of the mempool would give us everything we need - just spitballing, not sure if this would be an improvement

Reply:
Wasn't aware that we tried to keep the `core` package as close to geth as possible, will see if I can move this to somewhere in `plugin`.